decidim-elections/app/controllers/decidim/elections/elections_controller.rb

```diff
@@ -13,6 +13,11 @@ class ElectionsController < Decidim::Elections::ApplicationController
 
       def index
         redirect_to election_path(single, single: true) if single?
+
+        @scheduled_elections = search_klass.new(search_params.merge(state: %w(active upcoming)))
+        @finished_elections = search_klass.new(search_params.merge(state: %w(finished)))
+
+        @no_scheduled_elections = true if @scheduled_elections.results.empty? && @finished_elections.results.present?
```

Comment:
Maybe defining only a `scheduled_elections` helper method like the following would be simpler and easy to read?

```ruby
def scheduled_elections
  @scheduled_elections ||= search_klass.new(search_params.merge(state: %w(active upcoming))).results
end
```

Then, in the .erb, you could do something like:

```
if elections.none?
 # no elections message
elsif scheduled_elections.none?
 # no upcoming or active elections message
 # this only can be true if there is at last one published election and all are finished 
end
```

Do you think that this way is simpler (and still right 😅)?

Reply:
Haha, so much simpler and yes, still right! 😊